streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -584,6 +584,7 @@ public synchronized boolean hasNext() {
             this.rocksDBStore = rocksDBStore;
         }
 
+        @SuppressWarnings("unchecked")
```

Comment:
Why do we need this? Seems no unchecked warnings?

Reply:
IntelliJ does not show a warning, but `./gradlew` does -- I am also not sure why this warning appears in the build to be honest... It seems to be type safe code to me.

Reply:
yeah same here.. could you paste the warning message here if you still have it convienent?

Reply:
```
/Users/matthias/IdeaProjects/kafka/streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java:589: warning: [unchecked] unchecked call to restoreAllInternal(Collection<KeyValue<byte[],byte[]>>) as a member of the raw type RocksDBStore
            rocksDBStore.restoreAllInternal(records);
                                           ^
```

Reply:
Good catch. This change implies a lot of other changes, too... I added a commit with all those.